cf_xarray/helpers.py

```diff
@@ -211,11 +211,54 @@ def _bounds_helper(values, n_core_dims, nbounds, order):
             vertex_vals = np.block([[bot_left, bot_right], [top_left, top_right]])
     elif n_core_dims == 1 and nbounds == 2:
         # Middle points case (1D lat/lon)
-        vertex_vals = np.concatenate((values[..., :, 0], values[..., -1:, 1]), axis=-1)
+        vertex_vals = _get_ordered_vertices(values)
 
     return vertex_vals
 
 
+def _get_ordered_vertices(bounds: xr.DataArray) -> np.ndarray:
+    """Extracts a sorted 1D array of unique vertex values from a bounds DataArray.
+
+    This function takes a DataArray (or array-like) containing bounds information,
+    typically as pairs of values along the last dimension. It flattens the
+    bounds into pairs, extracts all unique vertex values, and returns them in
+    sorted order. The sorting order (ascending or descending) is determined by
+    inspecting the direction of the first non-equal bounds pair.
+
+    Parameters
+    ----------
+    bounds : xr.DataArray
+        A DataArray containing bounds information, typically with shape (..., 2).
+
+    Returns
+    -------
+    np.ndarray
+        A 1D NumPy array of sorted unique vertex values extracted from the
+        bounds.
+    """
+    # Convert to array if needed
+    arr = bounds.values if isinstance(bounds, xr.DataArray) else bounds
+    arr = np.asarray(arr)
+
+    # Flatten to (N, 2) pairs and get all unique values.
+    pairs = arr.reshape(-1, 2)
+    vertices = np.unique(pairs)
```

Comment:
Not sure I understand this `np.unique` ? If a coordinate is repeated, then the unique vector would be one element shorter then the full one, which means the `vertices` won't be of length N + 1 where N is the length of the coordinate.

I know repeated elements in coordinate aren't good, but I'm not sure here is the place to check that ?

Reply:
I removed the `np.unique()` and preserve repeated elements here [`30af8cc` (#579)](https://github.com/xarray-contrib/cf-xarray/pull/579/commits/30af8cc47e835b5377767f93967583230a09aa01).